NuRadioMC/utilities/muon_flux.py

```diff
@@ -0,0 +1,180 @@
+
+        Returns
+        -------
+        interpolator: function
+            function that returns the flux for a given energy
+        """
+        angle_edges = np.arccos(np.linspace(np.cos(theta_max), np.cos(theta_min), n_steps + 1))
+        angle_centers = 0.5 *(angle_edges[1:] + angle_edges[:-1])
+        d_cos_theta = np.abs(np.cos(theta_min) - np.cos(theta_max))
+
+        #print(f"integrating the flux from {theta_min/units.deg:.1f} deg to {theta_max/units.deg:.1f} deg by adding the flux from {angle_centers/units.deg}")
+
+        flux = None
+        for angle in angle_centers:
+            e_grid, flux_tmp = self.get_mu_flux(angle, altitude, primary_model=primary_model,
+                                            interaction_model=interaction_model, particle_names=particle_names)
+            
+            # solid angle element is sin(theta) dtheta dphi in spherical coordinates.
+            flux_tmp *= np.sin(angle) * (d_cos_theta * 2 * np.pi ) / n_steps 
+            if(flux is None):
+                flux = flux_tmp
+            else:
+                flux += flux_tmp
+
+        return interp1d(np.log10(e_grid), flux, kind='cubic')
+
+
+    def get_int_angle_mu_flux_buffered(self, energy, theta_min, theta_max, altitude=3200, n_steps=3, primary_model=(crf.GlobalSplineFitBeta, None),interaction_model='SIBYLL23C', particle_names=("total_mu+", "total_mu-")):
+
+        """
+        The function get_int_angle_mu_flux evalueates the integrated muon flux from theta_min to theta_max and caches the result.
+            
+        Parameters
+        ----------
+        energy: float
+            energy in eV
+        theta_min: float
+            minimum zenith angle in rad
+        theta_max: float
+            maximum zenith angle in rad
+        altitude: float
+            altitude in meters
+        n_steps: int
+            number of steps to use for the numerical integration
+
+        Returns
+        -------
+        flux: float
+            zenith angle integrated flux in NuRadioReco units 1/(area * time  * energy)   
+        """
+
+        params = (np.round(energy), np.round(theta_min, 6), np.round(theta_max, 6), np.round(altitude),
+                n_steps, primary_model, interaction_model, particle_names)
+
+        if(params not in self.__buffer):
+            print(f"calculating muon flux for {params}")
+            finterp = self.get_interp_angle_mu_flux(theta_min, theta_max, altitude, n_steps=n_steps, primary_model=primary_model,
+                                        interaction_model=interaction_model, particle_names=particle_names)
+
+            flux = finterp(np.log10(energy))
+            self.__buffer[params] = flux
+            with open(self.file_buffer, "wb") as fout:
+                pickle.dump(self.__buffer, fout, protocol=4)
+
+        return self.__buffer[params]
+    
+
+    def get_e_grid(self, theta_deg=50, interaction_model='SIBYLL23C', primary_model=(crf.GlobalSplineFitBeta, None)):
```

Comment:
the units seems to be wrong. All units should always be passed in the default units. If MCEq needsit in degree, just divide it by `units.deg` when you pass it to MCEq.

Reply:
Thanks for catching, I resolved it now.